msys2 build fixes

PeteHaughie opened the pull request:
Replaced the ppl.h requirement for building
Added build rules to addon_config
Added explanation to readme

PeteHaughie committed:
Replaced the ppl.h requirement for building
Added uild rules to addon_config
Added explanation to readme

bangnoise commented:
Thanks for this - I've made some changes which enable building with msys2. I didn't go with your version because

1. you can achieve a more concise msys2 section in addon_config.mk by using `ADDON_PKG_CONFIG_LIBRARIES`
2. `std::async` makes no guarantees about thread lifetime - it's free to spawn a new thread for every iteration, making it very inefficient. I've used `tbb::parallel_for` as it's available under MSYS2 and we are already using it for linux.

Hopefully this works for you - let me know if not. If you're on OpenFrameworks nightly builds you will need to use the `of_head` branch of this repo which I've just updated.

Thanks for the provocation to take action.

PeteHaughie commented:
All good. As I said before I'm new to C++ so I'm willing to take a backseat
to the experienced.

I'm just very glad this really quite excellent addon is back up and running.

On Wed, 12 Mar 2025, 1:47 am Tom Butterworth, ***@***.***>
wrote:

> Thanks for this - I've made some changes which enable building with msys2.
> I didn't go with your version because
>
>    1. you can achieve a more concise msys2 section in addon_config.mk by
>    using ADDON_PKG_CONFIG_LIBRARIES
>    2. std::async makes no guarantees about thread lifetime - it's free to
>    spawn a new thread for every iteration, making it very inefficient. I've
>    used tbb::parallel_for as it's available under MSYS2 and we are
>    already using it for linux.
>
> Hopefully this works for you - let me know if not. If you're on
> OpenFrameworks nightly builds you will need to use the of_head branch of
> this repo which I've just updated.
>
> Thanks for the provocation to take action.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bangnoise/ofxHapPlayer/pull/82#issuecomment-2716080905>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMR2232OK6MW2HJEYPVXID2T572BAVCNFSM6AAAAABYXPKJ3WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJWGA4DAOJQGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: bangnoise]*bangnoise* left a comment (bangnoise/ofxHapPlayer#82)
> <https://github.com/bangnoise/ofxHapPlayer/pull/82#issuecomment-2716080905>
>
> Thanks for this - I've made some changes which enable building with msys2.
> I didn't go with your version because
>
>    1. you can achieve a more concise msys2 section in addon_config.mk by
>    using ADDON_PKG_CONFIG_LIBRARIES
>    2. std::async makes no guarantees about thread lifetime - it's free to
>    spawn a new thread for every iteration, making it very inefficient. I've
>    used tbb::parallel_for as it's available under MSYS2 and we are
>    already using it for linux.
>
> Hopefully this works for you - let me know if not. If you're on
> OpenFrameworks nightly builds you will need to use the of_head branch of
> this repo which I've just updated.
>
> Thanks for the provocation to take action.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bangnoise/ofxHapPlayer/pull/82#issuecomment-2716080905>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMR2232OK6MW2HJEYPVXID2T572BAVCNFSM6AAAAABYXPKJ3WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMJWGA4DAOJQGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
